src/Hyperion/Analysis.hs

```diff
@@ -49,11 +49,13 @@ namesOf = go []
     coerce = contramap (const ()) . fmap (const ())
 
 analyze
-  :: Text -- ^ Benchmark name
-  -> Sample -- ^ Measurements
+  :: Text -- ^ Executable name.
+  -> Text -- ^ Benchmarks name.
+  -> Text -- ^ Benchmark name.
+  -> Sample -- ^ Measurements.
   -> Report
-analyze name samp = Report
-    { _reportBenchName = name
+analyze executableName benchesName name samp = Report
+    { _reportBenchName = Text.concat [executableName, ".", benchesName, ".", name]
```

Comment:
Benchmark name is the name passed to `defaultMain`.

As you remarked before, a user can call this function multiple times from a same benchmark executable and can also have multiple benchmark executables in their project.

Taking the two allows to avoid collisions.